src/yaml/any.cr

```diff
@@ -296,16 +314,16 @@ struct YAML::Any
     @raw.pretty_print(pp)
   end
 
-  # Returns `true` if both `self` and *other*'s raw object are equal.
-  def ==(other : YAML::Any)
-    raw == other.raw
-  end
-
   # Returns `true` if the raw object is equal to *other*.
   def ==(other)
     raw == other
   end
 
+  # Returns `true` if both `self` and *other*'s raw object are equal.
+  def ==(other : YAML::Any)
+    raw == other.raw
+  end
```

Comment:
Why was the order filpped?

Reply:
I was experimenting with the error messages. Before, when I've done a catch-all and a more specific overload of a method, I've had to put the catch-all version first or the catch-all would override the more specific overload. I don't know if this was me misunderstanding the type system at the time (it was a few years ago) and misattributing the error, but either way that's apparently not how it works anymore, if it ever did.

<details>

```crystal
def foo(bar : String)
  "strings only"
end

def foo(bar)
  "catch all"
end

def foo(bar : Int)
  123
end

foo 123 # => 123

foo "abc" # => "strings only"

foo %w[abc] # => "catch all"
```

</details>